packages/office-ui-fabric-react/src/components/List/List.tsx

```diff
@@ -294,58 +299,94 @@ export class List extends BaseComponent<IListProps, IListState> implements IList
+      getPageClassName
+    } = this.props;
+    const {
+      pages = []
+    } = this.state;
+    let pageElements: HTMLElement[] = [];
     let divProps = getNativeProps(this.props, divProperties);
 
-    // assign list if no role
-    role = (role === undefined) ? 'list' : role;
-
-    for (let i = 0; i < pages!.length; i++) {
-      pageElements.push(this._renderPage(pages![i]));
+    for (const page of pages) {
+      pageElements.push(onRenderPage({
+        page: page,
+        role: 'presentation',
+        className: css('ms-List-page', getPageClassName && getPageClassName(page)),
+        key: page.key,
+        ref: page.key,
+        style: this._getPageStyle(page)
+      }, this._onRenderPage));
     }
 
     // console.log(`Page elements ${pageElements.length}`);
 
     return (
-      <div ref='root' { ...divProps } role={ role } className={ css('ms-List', className) } >
-        <div ref='surface' className='ms-List-surface' role='presentation'>
+      <div
+        ref='root'
+        { ...divProps }
+        role={ (role === undefined) ? 'list' : role }
+        className={ css('ms-List', className) }
+      >
+        <div ref='surface' className={ css('ms-List-surface', surfaceClassName) } role='presentation'>
           { pageElements }
         </div>
       </div>
     );
   }
 
-  private _renderPage(page: IPage): any {
-    let { onRenderCell, role } = this.props;
-    let cells = [];
-    let pageStyle = this._getPageStyle(page);
+  @autobind
+  private _onRenderPage(pageProps: IPageProps, defaultRender?: IRenderFunction<IPageProps>): any {
+    const {
+      onRenderCell,
+      role,
+      getCellStyle,
+      getCellClassName
+    } = this.props;
+
+    const {
+      page,
+      ...divProps
+    } = pageProps;
+
+    const {
+      items,
+      startIndex
+    } = page;
 
     // only assign list item role if no role is assigned
-    role = (role === undefined) ? 'listitem' : 'presentation';
+    const cellRole = (role === undefined) ? 'listitem' : 'presentation';
 
-    for (let i = 0; page.items && i < page.items.length; i++) {
-      let item = page.items[i];
-      const index = page.startIndex + i;
-      let itemKey =
-        this.props.getKey
-          ? this.props.getKey(item, index)
-          : item && item.key;
+    const cells = (items || []).map((item: any, offset: number) => {
+      const index = startIndex + offset;
+
+      let itemKey = this.props.getKey ?
+        this.props.getKey(item, index) :
+        item && item.key;
 
       if (itemKey === null || itemKey === undefined) {
         itemKey = index;
       }
 
-      cells.push(
-        <div role={ role } className='ms-List-cell' key={ itemKey } data-list-index={ i + page.startIndex } data-automationid='ListCell'>
-          { onRenderCell!(item, page.startIndex + i) }
+      return (
+        <div role={ cellRole }
+          className={ css('ms-List-cell', getCellClassName && getCellClassName(item, index)) }
```

Comment:
I'm afraid that 'cell' is already in the current domain language, given the existence of the `onRenderCell` field.